File: node/models/transportModel.js
import mongoose from "mongoose";

const transportSchema = new mongoose.Schema(
  {
    id: { type: String, required: true },
    name: { type: String, required: true },
    location: { type: String },
    images: [{ type: String }], // ✅ مصفوفة صور
    rating: { type: Number, default: 0 },
    description: { type: String },
    type: { type: String },
    fare: { type: Number, default: 0 },
    phoneNumber: { type: String }, // اختياري
  },
  { timestamps: true }
);

export default mongoose.model("Transport", transportSchema);

File: traveling_app/lib/models/transport_model.dart
import 'place_model.dart';

class TransportModel extends PlaceModel {
  final String type; // مثلاً: سيارة أجرة - حافلة - مترو
  final double fare;

  TransportModel({
    required super.id,
    required super.name,
    required super.description,
    required super.images, // ← صورة واحدة داخل قائمة
    required super.location,
    required super.rating,
    required this.type,
    required this.fare,
  });
}
